src/renderer/components/+cluster-settings/components/cluster-home-dir-setting.tsx

```diff
@@ -1,86 +1,44 @@
 import React from "react";
-import { Cluster } from "../../../../main/cluster";
-import { Input } from "../../input";
-import { Spinner } from "../../spinner";
-import { clusterStore } from "../../../../common/cluster-store"
-import { Icon } from "../../icon";
-import { Tooltip, TooltipPosition } from "../../tooltip";
-import { autobind } from "../../../utils";
-import { TextInputStatus } from "./statuses"
+import throttle from "lodash/throttle";
 import { observable } from "mobx";
 import { observer } from "mobx-react";
+import { Cluster } from "../../../../main/cluster";
+import { Input } from "../../input";
+import { SubTitle } from "../../layout/sub-title";
 
 interface Props {
-    cluster: Cluster;
+  cluster: Cluster;
 }
 
 @observer
 export class ClusterHomeDirSetting extends React.Component<Props> {
   @observable directory = this.props.cluster.preferences.terminalCWD || "";
-  @observable status = TextInputStatus.CLEAN;
-  @observable errorText?: string;
 
-  render() {
-    return <>
-      <h4>Working Directory</h4>
-      <p>Set initial working directory for terminals. When set it will the `pwd` when a new terminal instance is opened for this cluster.</p>
-      <Input
-        theme="round-black"
-        className="box grow"
-        value={this.directory}
-        onSubmit={this.onWorkingDirectorySubmit}
-        onChange={this.onWorkingDirectoryChange}
-        iconRight={this.getIconRight()}
-        placeholder="$HOME"
-      />
-    </>;
-  }
+  save = throttle((value: string) => {
+    this.props.cluster.preferences.terminalCWD = value;
+  }, 500);
 
-  @autobind()
-  onWorkingDirectoryChange(directory: string, _e: React.ChangeEvent) {
-    if (this.status === TextInputStatus.UPDATING) {
-      console.log("prevent changing cluster directory while updating");
-      return;
-    }
-    
-    this.status = this.dirDiffers(directory);
-    this.directory = directory;
+  onChange = (value: string) => {
+    this.directory = value;
+    this.save(value);
```

Comment:
Yes, it can happen. In theory, if updating would happened rarely (for example if we construct some `Save` button) same situation can appear in case where user just saved wrong value. @nevalla Wdyt? Is there any background process can be harmed by quick parameter update? Now it updates itself every 500ms while typing.

Reply:
Can we save when the field loses the focus (`onblur`)?

Reply:
Sure, that would be best decision 👍

Reply:
Why not also on submit (enter) as well?